Server/controllers/monitorController.js

```diff
@@ -214,6 +214,10 @@ const editMonitor = async (req, res, next) => {
 
   try {
     const editedMonitor = await req.db.editMonitor(req, res);
+    // Delete the old job(editedMonitor has the same ID as the old monitor)
+    req.jobQueue.deleteJob(editedMonitor);
```

Comment:
My suggestion is that we could use async/await more consistently throughout the function for cleaner code structure.

And also for a more specific error handling and haunting the error on a more exact spot, you may want to write this line like this:

```
// Delete the old job from the job queue
    try {
      await req.jobQueue.deleteJob(editedMonitor);
    } catch (error) {
      // Log the error and continue, as the monitor was still successfully edited
      console.error(some constant text to show the error type , error);
    }
```

Reply:
Good call, I forgot this was an async function, I will add the await.

The error handling is done further upstream in the express middleware, so throwing the error is sufficient here and the middleware will take care of it.

Reply:
Awesome 👍🏻✌️✅